File: Projekt/Backend/system-kliniki-medycznej/src/main/java/org/back/systemklinikimedycznej/doctor/controller/dto/DoctorDetails.java
package org.back.systemklinikimedycznej.doctor.controller.dto;

import org.back.systemklinikimedycznej.doctor.repositories.entities.DoctorOfficeHours;
import org.back.systemklinikimedycznej.personal_details.dto.PersonalDetailsDto;

import java.time.LocalDate;
import java.util.List;

public record DoctorDetails(
        String username,
        String email,
        String profileImagePath,
        PersonalDetailsDto personalDetails,
        String description,
        List<DoctorSpecializationDto> doctorSpecializations,
        List<OfficeHoursDto> doctorOfficeHours,
        String pwzNumber,
        LocalDate dateOfEmployment

) {
}

File: Projekt/Frontend/system-kliniki-medycznej/src/app/features/doctor/domain/doctor-details.ts
import {PersonalDetails} from '../../personal-details/domain/personal-details';
import {DoctorSpecialization} from './doctor-specialization';
import {OfficeHours} from './office-hours';

export interface DoctorDetails {
  username: string,
  email: string,
  personalDetails: PersonalDetails,
  description: string,
  doctorSpecializations: DoctorSpecialization[],
  doctorOfficeHours: OfficeHours[],
  pwzNumber: string,
  dateOfEmployment: string
}
